UI/src/router.js

```diff
@@ -7,6 +7,42 @@
       mapView.setZoom(zoom);
     };
 
+    var linearCentering = function(layerName, id){
+      applicationModel.selectLayer(layerName);
+      var asset = _(models.linearAssets).find({ layerName: layerName });
+      var linearAsset = asset.selectedLinearAsset.getLinearAsset(parseInt(id));
+      if (linearAsset) {
+        asset.selectedLinearAsset.open(linearAsset, true);
+        applicationModel.setSelectedTool('Select');
+      }
+      backend.getLinearAssetMidPoint(asset.typeId, id).then(function (result) {
```

Comment:
maintenanceRoad already has a request to get the mid point, probably we should use the same.